### DISCUSSION TITLE
How to explore the options of field inside list.

### DISCUSSION THREAD

**Discussion by gopaldahal** [pnp/pnpcore#505] [General]
I didn't find any object that gives the information on field options.

For Eg, let say we are creating a field of Number type inside the list
```
IField myField = await myList.Fields.AddNumberAsync("My Field", new FieldNumberOptions()
{
    Group = "Custom Fields",
    AddToDefaultView = true,
    MinimumValue = 0,
    MaximumValue = 100
});
```
While playing with the items inside the list, It may need to find the FieldNumberOptions meta information like range (MinimumValue , MaximumValue ) of that particular field.

Was expecting myList.Fields["My Field"].FieldOptions type of object for the same.

_Thanks for the contribution! Sharing is Caring!_

**Comment by jansenbe** [comment]
@gopaldahal : When you load fields, you'll get back a collection of `IField` objects. Depending on the type of the field (`FieldTypeKind` property) you can verify if the properties you need are loaded or not (e.g. `myfield.IsPropertyAvailable(p=>p.MinimumValue`) and if so, use the property.

An alternative approach can be used using the field schema XML: when creating the fields, we take the configuration settings (e.g. `FieldNumberOptions`) and translate that the field schema XML (CAML) as the only flexible method to create fields depends on providing the field schema. The same applies when you read fields, the APIs will return the field schema XML. A viable option would be that you parse the schema XMLs for the fields you're interested in and grab the min/max values from that. The properties in the various `FieldxxxOptions` classes match with the properties used in the Field schema XML.
